src/main/java/backend/resource/Model.java

```diff
@@ -99,7 +103,7 @@ public UpdateSignature getUpdateSignature() {
     }
 
     @Override
-    public List<TurboIssue> getIssues() {
+    public synchronized List<TurboIssue> getIssues() {
```

Comment:
The `UpdateModelTask` calls `getIssues` on a `Model` to do a reconciliation, while at the same time, we can also do `replaceIssueLabels` on that `Model`. So this would help to some extent in that case. The thing that breaks this is that after the `getIssues`, we modify the returned list of issues directly without the synchronisation protection from `Model`.